File: python/quicksort.py
from random import randrange

def quicksort(arr: list[int]) -> list:
	if len(arr) < 2:
		return arr
	pivot = arr.pop(randrange(len(arr)))
	left = [num for num in arr if num <= pivot]
	right =[num for num in arr if num > pivot]
		
	return quicksort(left) + [pivot] + quicksort(right)

File: javascript/quicksort.js
function quicksort(arr){
	if(arr.length < 2)
		return arr;

	let pivot = arr.splice(Math.floor(Math.random() * arr.length),1)[0];
	let left = [];
	let right = [];
	for (let i = 0; i < arr.length; i++){
		if(pivot > arr[i])
			left.push(arr[i]);
		else
			right.push(arr[i]);
	}
		
	return [...quicksort(left), pivot, ...quicksort(right)];
}